tests/controllers/portfolios.controller.test.ts

```diff
@@ -0,0 +1,274 @@
+import PortfoliosController from '../../src/controllers/portfolios.controller'
+import portfolioService from '../../src/services/portfolio.service'
+import { IPortfolio } from '../../src/models/portfolio';
+import { testPortfolios } from '../utils/mockData'
+
+describe("Users Controller", () => {  
+    let portfolioController: PortfoliosController;
+    let res: any;
+    let nextFunction: any;
+
+    beforeAll(() => {
+        portfolioController = new PortfoliosController(portfolioService);
+        res = {
+            send: jest.fn()
+        };
+        nextFunction = jest.fn();
+    });
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+    });
+    
+    it("gets all portfolios for the current user", async () => { 
+        portfolioService.getAllByFields = jest.fn().mockImplementation(() => {
```

Comment:
so this mockImplementation when it gets called it automatically mocks the implementation with the code you added?